### ISSUE TITLE
Redis hang workflow when throws an error

### ISSUE THREAD

**Issue by AllanDaemon** [n8n-io/n8n#1709]
**Describe the bug**
When the Redis node tries to set a value that is null, it fails. This failure instead of giving an error, it hangs the n8n.

**To Reproduce**
Use the simple workflow:

```json
{
  "nodes": [
    {
      "parameters": {},
      "name": "Start",
      "type": "n8n-nodes-base.start",
      "typeVersion": 1,
      "position": [
        250,
        300
      ]
    },
    {
      "parameters": {
        "functionCode": "item.test = null;\nreturn item;"
      },
      "name": "FunctionItem",
      "type": "n8n-nodes-base.functionItem",
      "typeVersion": 1,
      "position": [
        450,
        300
      ]
    },
    {
      "parameters": {
        "operation": "set",
        "key": "test",
        "value": "={{ $json.test }}",
        "expire": true
      },
      "name": "Redis",
      "type": "n8n-nodes-base.redis",
      "typeVersion": 1,
      "position": [
        650,
        300
      ],
      "credentials": {
        "redis": "Redis Local"
      }
    }
  ],
  "connections": {
    "Start": {
      "main": [
        [
          {
            "node": "FunctionItem",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "FunctionItem": {
      "main": [
        [
          {
            "node": "Redis",
            "type": "main",
            "index": 0
          }
        ]
      ]
    }
  }
}
```
**Expected**

I understand that Redis cannot properly serialize the `null` value, so I expect it to generate an error. And it seems that the code throws an error, but something in the implementation about handling exceptions in async functions or promises isn't right. This is the actual bug, because instead of the error bubble up, and showing in the UI, it creates a message in the console and the workflow hangs forever. This is what is print to the console:

```
(node:379373) UnhandledPromiseRejectionWarning: TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)                                                                                      
    at setValue (/home/rea/n8n/node_modules/n8n-nodes-base/dist/nodes/Redis/Redis.node.js:376:42)
    at RedisClient.<anonymous> (/home/rea/n8n/node_modules/n8n-nodes-base/dist/nodes/Redis/Redis.node.js:462:35)
    at RedisClient.emit (events.js:315:20)
    at RedisClient.on_ready (/home/rea/n8n/node_modules/redis/index.js:422:10)
    at RedisClient.on_info_cmd (/home/rea/n8n/node_modules/redis/index.js:450:18)
    at /home/rea/n8n/node_modules/redis/index.js:471:14
    at Object.callbackOrEmit [as callback_or_emit] (/home/rea/n8n/node_modules/redis/lib/utils.js:89:9)
    at Command.callback (/home/rea/n8n/node_modules/redis/lib/individualCommands.js:157:15)
    at normal_reply (/home/rea/n8n/node_modules/redis/index.js:655:21)
    at RedisClient.return_reply (/home/rea/n8n/node_modules/redis/index.js:753:9)
    at JavascriptRedisParser.returnReply (/home/rea/n8n/node_modules/redis/index.js:138:18)
    at JavascriptRedisParser.execute (/home/rea/n8n/node_modules/redis-parser/lib/parser.js:544:14)
    at Socket.<anonymous> (/home/rea/n8n/node_modules/redis/index.js:219:27)
    at Socket.emit (events.js:315:20)
    at addChunk (internal/streams/readable.js:309:12)
    at readableAddChunk (internal/streams/readable.js:284:9) 
    at Socket.Readable.push (internal/streams/readable.js:223:10)
    at TCP.onStreamRead (internal/stream_base_commons.js:188:23)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:379373) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To termijs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:379373) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
And this is printed when you ask to close the process.
```
Waiting for 1 active executions to finish...
 - Execution ID 5889, workflow ID: 29
```
After restarted, the executions list show this:
![image](https://user-images.githubusercontent.com/607994/116156261-9d20bf00-a6c1-11eb-9dfd-9c7fc240fd6e.png)

**Environment:**
 - OS: `Ubuntu 20.04.2`
 - n8n Version: `0.116.1`
 - Node.js Version: `v14.16.1`
 - `EXECUTIONS_PROCESS=main`

**Comment by RicardoE105** [comment]
@AllanDaemon got fixed. We will let you know when it is released.

**Comment by janober** [comment]
Got released with n8n@0.118.0
